frontend/ui/widget/inputtext.lua

```diff
@@ -664,6 +670,42 @@ function InputText:onKeyPress(key)
             handled = false
         end
     end
+    if Device:hasSymKey() then
+        if not handled and key["Sym"] and not key["Shift"] and not key["Alt"] then
+            handled = true
+            if     key["Q"] then self:addChars("!")
+            elseif key["W"] then self:addChars("?")
+            elseif key["E"] then self:addChars("-")
+            elseif key["R"] then self:addChars("_")
```

Comment:
You'll have to replace that by a table lookup so its cheaper than all these tests.
Something like this as a module local:
```lua
local sym_key_map = {
  "Q" = "!",
  "W" = "?",
...
}
```
and here:

```lua
    local symkey = sym_key_map[key.key] -- looks like the main key is available as key.key
    if symkey then
        self:addChars(symkey)
    else
        handled = false
    end
```

Reply:
where do I put it?

Reply:
`local sym_key_map =` ? Near, above `function InputText:onKeyPress` and its comment.

Reply:
that's what I thought, yet 

```
./luajit: frontend/ui/widget/textviewer.lua:23: loop or previous error loading module 'ui/widget/inputdialog'
stack traceback:
	[C]: in function 'require'
	frontend/ui/widget/textviewer.lua:23: in main chunk
	[C]: in function 'require'
	frontend/document/documentregistry.lua:294: in main chunk
	[C]: in function 'require'
	frontend/ui/widget/imagewidget.lua:134: in function '_loadfile'
	frontend/ui/widget/imagewidget.lua:262: in function '_render'
	frontend/ui/widget/imagewidget.lua:380: in function 'getSize'
	frontend/ui/widget/horizontalgroup.lua:24: in function 'getSize'
	frontend/ui/widget/container/framecontainer.lua:54: in function 'getSize'
	frontend/ui/widget/infomessage.lua:180: in function 'init'
	frontend/ui/widget/widget.lua:46: in function 'new'
	frontend/userpatch.lua:81: in function 'runUserPatchTasks'
	frontend/userpatch.lua:98: in function 'applyPatches'
	./reader.lua:184: in main chunk
	[C]: at 0x00015038
lipc-wait-event exited normally with status: 0
```

Reply:
`frontend/userpatch.lua:81: in function 'runUserPatchTasks'`

Reply:
> `frontend/userpatch.lua:81: in function 'runUserPatchTasks'`

what does that mean?

Reply:
It means this crash is caused by one of your userpatches.
Disable them to see if it works without - and then try to find and fix the userpatch that provokes it.

Reply:
there is only one patch (the one that switches the dictionary with wiki.. I think)

anyway now I get this:

```
./luajit: error loading module 'ui/widget/inputtext' from file 'frontend/ui/widget/inputtext.lua':
	frontend/ui/widget/inputtext.lua:590: '}' expected (to close '{' at line 589) near '='
stack traceback:
	[C]: at 0x00070f14
	[C]: in function 'require'
	frontend/ui/widget/inputdialog.lua:109: in main chunk
	[C]: in function 'require'
	frontend/ui/widget/textviewer.lua:23: in main chunk
	[C]: in function 'require'
	frontend/document/documentregistry.lua:294: in main chunk
	[C]: in function 'require'
	frontend/apps/reader/readerui.lua:13: in main chunk
	[C]: in function 'require'
	./reader.lua:262: in main chunk
	[C]: at 0x00015038
lipc-wait-event exited normally with status: 0
```

```lua
local sym_key_map = {
    "Q" = "!", "W" = "?", "E" = "-", "R" = "_", "T" = "%", "Y" = "=", "U" = "7", "I" = "8",  "O" = "9", "P" = "0",
    "A" = "<", "S" = ">", "D" = "(", "F" = ")", "G" = "#", "H" = "'", "J" = "4", "K" = "5",  "L" = "6",
    "Z" = "{", "X" = "}", "C" = "[", "V" = "]", "B" = "1", "N" = "2", "M" = "3", "." = ":", "AA" = ";",
}
```

Reply:
Sorry for my bad suggestion, but don't trust me and look around.
The corect syntax seems to be:
```lua
local sym_key_map = {
  ["Q"] = "!",
  ["W"] = "?",
}
```